src/main/java/bio/terra/common/BlockingRejectedExecutionHandler.java

```diff
@@ -0,0 +1,55 @@
+package bio.terra.common;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+
+/**
+ * An implementation of {@link RejectedExecutionHandler}.
+ *
+ * <p>If a {@link ThreadPoolExecutor} can't accept a task due to saturation of its threadpool and
+ * queue, this handler blocks when waiting for a spot to become available in the queue rather than
+ * rejecting the task.
+ */
+public class BlockingRejectedExecutionHandler implements RejectedExecutionHandler {
+  @VisibleForTesting
+  static String interruptedExceptionMessage(Runnable r, ThreadPoolExecutor executor) {
+    return "Task %s interrupted while waiting to be added to queue: %s".formatted(r, executor);
```

Comment:
Does it make sense to use `Runnable.toString()` here? By default this is the same as `Object.toString()` (which would print the object's type and memory address), but maybe the `Runnable`s used in TDR will print something more useful.

Reply:
Good shout -- I don't think this is useful, so I'll remove it.  Conversely, the `ThreadPoolExecutor.toString()` method is very useful so I'll leave that in:

https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/util/concurrent/ThreadPoolExecutor.java#L1856-L1891